buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1095,7 +1084,6 @@ public int writeBytes(InputStream in, int length)
 
     @Override
     public int writeBytes(ScatteringByteChannel in, int length) throws IOException {
-        ensureAccessible();
         ensureWritable(length);
         int writtenBytes = setBytes(writerIndex, in, length);
```

Comment:
Discussed with @normanmaurer offline ... both methods are public so we would have to create additional methods that bypass the additional call to ensureWritable (e.g `writeBytes` calls `setBytes0` which calls `checkIndex0`) ... lets not worry about this for now.